api/handlers.go

```diff
@@ -31,3 +32,123 @@ func (api *API) userHandlerGET(w http.ResponseWriter, req *http.Request, _ httpr
 	}
 	WriteJSON(w, u)
 }
+
+// userUploadsHandler returns all uploads made by the current user.
+func (api *API) userUploadsHandler(w http.ResponseWriter, req *http.Request, _ httprouter.Params) {
+	sub, _, _, err := tokenFromContext(req)
+	if err != nil {
+		WriteError(w, err, http.StatusInternalServerError)
+		return
+	}
+	u, err := api.staticDB.UserBySub(req.Context(), sub)
+	// TODO Should we create the user in case it's not found?
+	if err != nil {
+		WriteError(w, err, http.StatusInternalServerError)
+		return
+	}
+	ups, err := api.staticDB.UploadsByUser(req.Context(), *u)
+	if err != nil {
+		WriteError(w, err, http.StatusInternalServerError)
```

Comment:
This would/could be a failed dependency perhaps?

Reply:
```
424 Failed Dependency (WebDAV)
    The request failed due to failure of a previous request.
```
I don't think there is a previous request that has failed here.

All 4xx responses indicate that the service is in good health and the problem is on the user's side. So, they should either fix their request (400), get authenticated (401), get more rights (403), make sure they are trying to get an existing resource (404), try again later (429) or something else that's in their control. The failed dependency would mean "make sure your previous requests pass and try again". What we have here is our internal infrastructure preventing the request from being fulfilled. This is unlikely to be fixed by actions on the user's side, so it should be a 5xx.

Reply:
What is the case where `err` would not be `nil`? Database down for example?

Reply:
Any DB error or an invalid user. The latter one shouldn't happen because of the auto-create mechanic we're using.